src/ViewLayer/RaceModeDisplay/RaceModeDashboardView.cpp

```diff
@@ -187,15 +189,89 @@ void RaceModeDashboardView::initalizeFaultAnimation()
 
 void RaceModeDashboardView::runFaultAnimation()
 {
-    int numberOfActiveFaults = batteryFaultsList_.numberOfActiveFaults() +
-                               motorOneFaultsList_.numberOfActiveFaults() +
-                               motorZeroFaultsList_.numberOfActiveFaults();
+    QVector<QString> batteryFaults = batteryFaultsList_.activeFaultLabels();
+    QVector<QString> motorZeroFaults = motorZeroFaultsList_.activeFaultLabels();
+    QVector<QString> motorOneFaults = motorOneFaultsList_.activeFaultLabels();
 
-    if (numberOfActiveFaults > 0 && faultAnimation_->state() != QAbstractAnimation::Running)
+    if (faultAnimation_->state() != QAbstractAnimation::Running && triggerFaultAnimation(batteryFaults, motorZeroFaults, motorOneFaults))
     {
         faultAnimation_->setDirection(QAbstractAnimation::Forward);
         faultAnimation_->start();
     }
+
+    prevBatteryFaults_ = batteryFaults;
+    prevMotorZeroFaults_ = motorZeroFaults;
+    prevMotorOneFaults_ = motorOneFaults;
+}
+
+bool RaceModeDashboardView::triggerFaultAnimation(QVector<QString>& batteryFaults, QVector<QString>& motorZeroFaults, QVector<QString>& motorOneFaults)
+{
+
+//Will remove this comment right before merging
+//    QString str;
+//    for (int i = 0; i < batteryFaults.size(); ++i)
+//    {
+//        if (i > 0)
+//            str += "|||||";
+//        str += batteryFaults[i];
+//    }
+//    qDebug() << str << "\n";
+
+//    str = "";
+//    for (int i = 0; i < prevBatteryFaults_.size(); ++i)
+//    {
+//        if (i > 0)
+//            str += "|||||";
+//        str += prevBatteryFaults_[i];
+//    }
+//    qDebug() << str << "\n";
+
+    if (batteryFaults != prevBatteryFaults_ && !batteryFaults.empty() && !std::includes(prevBatteryFaults_.begin(), prevBatteryFaults_.end(), batteryFaults.begin(), batteryFaults.end()))
+    {
+        if (!prevBatteryFaults_.empty() && prevBatteryFaults_.size() > batteryFaults.size())
```

Comment:
Can we do this?

if (prevBatteryFaults_.empty() || prevBatteryFaults_.size() < batteryFaults.size())
{
   return true;
}

if (batteryFaults.back() != prevBatteryFaults_.at(batteryFaults.size() - 1))
{
   return true;
}


Here's my reasoning: We remove a nested if statement, and we remove the else section, leading to less lines of code, leading to easier readability.

Reply:
Who are you who is wise in these ways? I've also added this change.